e2e/src/pages/navigate.po.ts

```diff
@@ -1,20 +1,32 @@
-import { by, element } from 'protractor';
-import { AppPage } from './app.po';
+import {browser, by, element, ElementFinder} from 'protractor';
+import {AppPage} from './app.po';
 
-export class NavigatePage extends AppPage{
+export class NavigatePage extends AppPage {
+  public pageNumber: ElementFinder;
+  public next: ElementFinder;
+  public previous: ElementFinder;
 
-  async goToNextPage() {
-    await this.waitForElement(by.id('next'));
-    await this.clickElement(by.id('next'));
-  }
-
-  async goToPreviousPage() {
-    await this.waitForElement(by.id('previous'));
-    await this.clickElement(by.id('previous'));
+  constructor() {
+    super();
+    this.next = element(by.id('next'));
+    this.previous = element(by.id('previous'));
+    this.pageNumber = element(by.id('pageNumber'));
   }
 
   async number() {
     await this.waitForElement(by.id('pageNumber'));
     return element(by.id('pageNumber')).getAttribute('value');
   }
+
+  async goToNextPage() {
+    await this.next.click().then(async function () {
```

Comment:
I'm confused by this, you're mixing .then and await?

Reply:
fixed